cpp/include/raft/sparse/cusparse_wrappers.h

```diff
@@ -182,21 +182,337 @@ inline void cusparsecoosortByRow(  // NOLINT
+  return cusparseSpMV_bufferSize(handle, opA, alpha, matA, vecX, beta, vecY,
+                                 CUDA_R_32F, alg, bufferSize);
+}
+template <>
+inline cusparseStatus_t cusparsespmv_buffersize(
+  cusparseHandle_t handle, cusparseOperation_t opA, const double* alpha,
+  const cusparseSpMatDescr_t matA, const cusparseDnVecDescr_t vecX,
+  const double* beta, const cusparseDnVecDescr_t vecY, cusparseSpMVAlg_t alg,
+  size_t* bufferSize, cudaStream_t stream) {
+  CUSPARSE_CHECK(cusparseSetStream(handle, stream));
+  return cusparseSpMV_bufferSize(handle, opA, alpha, matA, vecX, beta, vecY,
+                                 CUDA_R_64F, alg, bufferSize);
+}
+
+template <typename T>
+cusparseStatus_t cusparsespmv(cusparseHandle_t handle, cusparseOperation_t opA,
+                              const T* alpha, const cusparseSpMatDescr_t matA,
+                              const cusparseDnVecDescr_t vecX, const T* beta,
+                              const cusparseDnVecDescr_t vecY,
+                              cusparseSpMVAlg_t alg, T* externalBuffer,
+                              cudaStream_t stream);
+template <>
+inline cusparseStatus_t cusparsespmv(
+  cusparseHandle_t handle, cusparseOperation_t opA, const float* alpha,
+  const cusparseSpMatDescr_t matA, const cusparseDnVecDescr_t vecX,
+  const float* beta, const cusparseDnVecDescr_t vecY, cusparseSpMVAlg_t alg,
+  float* externalBuffer, cudaStream_t stream) {
+  CUSPARSE_CHECK(cusparseSetStream(handle, stream));
+  return cusparseSpMV(handle, opA, alpha, matA, vecX, beta, vecY, CUDA_R_32F,
+                      alg, externalBuffer);
+}
+template <>
+inline cusparseStatus_t cusparsespmv(
+  cusparseHandle_t handle, cusparseOperation_t opA, const double* alpha,
+  const cusparseSpMatDescr_t matA, const cusparseDnVecDescr_t vecX,
+  const double* beta, const cusparseDnVecDescr_t vecY, cusparseSpMVAlg_t alg,
+  double* externalBuffer, cudaStream_t stream) {
+  CUSPARSE_CHECK(cusparseSetStream(handle, stream));
+  return cusparseSpMV(handle, opA, alpha, matA, vecX, beta, vecY, CUDA_R_64F,
+                      alg, externalBuffer);
+}
+/** @} */
+#else
+/**
+ * @defgroup Csrmv cusparse csrmv operations
+ * @{
+ */
+template <typename T>
+cusparseStatus_t cusparsecsrmv(  // NOLINT
+  cusparseHandle_t handle, cusparseOperation_t trans, int m, int n, int nnz,
+  const T* alpha, const cusparseMatDescr_t descr, const T* csrVal,
+  const int* csrRowPtr, const int* csrColInd, const T* x, const T* beta, T* y,
+  cudaStream_t stream);
+template <>
+inline cusparseStatus_t cusparsecsrmv(
+  cusparseHandle_t handle, cusparseOperation_t trans, int m, int n, int nnz,
+  const float* alpha, const cusparseMatDescr_t descr, const float* csrVal,
+  const int* csrRowPtr, const int* csrColInd, const float* x, const float* beta,
+  float* y, cudaStream_t stream) {
+  CUSPARSE_CHECK(cusparseSetStream(handle, stream));
+  return cusparseScsrmv(handle, trans, m, n, nnz, alpha, descr, csrVal,
```

Comment:
Looks like this one is deprecated in CUDA 11.0 (along with a few others). 
Are we addressing this in this PR or separately?

https://docs.nvidia.com/cuda/cusparse/index.html

Reply:
It's not clear what this comment refers to: `cusparseSetStream()` is being deprecated (it doesn't seem that's the case, based on the documentation linked above); or, `cusparse<X>csrmv()` ? (which is taken care of: see the `#if ` (line 216) and `#else` (line 249).

Reply:
My bad, I did miss the`#if` earlier in the file